File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int maxArea = Integer.MIN_VALUE;
        int pL = 0;
        int pR = height.length-1;
        while(pL!=pR){
            maxArea = Math.max(maxArea,Math.min(height[pL],height[pR])*(pR-pL));
            if(height[pL]<height[pR]){
                pL++;
            }else{
                pR--;
            }
        }
        return maxArea;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int pL {0};
        int pR = height.size()-1;
        int max {0};
        while(pL<pR){
            int maxHeight = height[pR]>height[pL] ? height[pL]:height[pR];
            int area = ((pR-pL) * maxHeight);
            max =  area>max ? area:max;
            if(height[pL]>height[pR]){
                pR--;
            }else{
                pL++;
            }
        }
        return max;
    }
};